### ISSUE TITLE
Snapshot.changedAttributes breaks for record-less snapshots in 3.14+

### ISSUE THREAD

**Issue by runspired** [emberjs/data#7015]
We (smartly) avoided allocating anything for changed attributes for the case where no record exists for the snapshot for perf savings; however, this has an unintended side-effect for users whom are serializing related records which potentially have not been materialized.

For these users, calling `snapshot.changedAttributes` will error with the following:

```
Uncaught (in promise) TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at Snapshot.changedAttributes (vendor.js:120369)
```

An easy fix for this would be for us to early exit with an empty object when we are in this case.

**Comment by gvocale** [comment]
Any news on this? It is blocking us from updating from 3.13 to newer versions

**Comment by igorT** [comment]
I have opened a PR to fix https://github.com/emberjs/data/pull/7080 should be released in 3.16 as point fix

**Comment by gvocale** [comment]
@igorT great! Thank you!
